components/VC/common/VCUtils.tsx

```diff
@@ -248,65 +257,106 @@ export const getMosipLogo = () => {
 /**
  *
  * @param wellknown (either supportedCredential's wellknown or whole well known response of issuer)
- * @param idType
- * @returns id Type translations (Eg - National ID)
+ * @param credentialConfigurationId
+ * @returns credential type translations (Eg - National ID)
  *
- * supportedCredential's wellknown is passed from getActivityText after fresh download
+ * supportedCredential's wellknown is passed from getActivityText after fresh download and viewing of card's mini view flow
  * & all other consumers pass whole well known response of issuer
  */
-export const getIdType = (
-  wellknown: CredentialTypes | IssuerWellknownResponse,
+export const getCredentialType = (
+  supportedCredentialsWellknown: CredentialTypes,
+): string => {
+  if (supportedCredentialsWellknown['display']) {
+    const idTypeObj = supportedCredentialsWellknown.display.map(
+      (displayProps: any) => ({
+        language: displayProps.locale,
+        value: displayProps.name,
+      }),
+    );
+    return getLocalizedField(idTypeObj);
+  }
+  if (supportedCredentialsWellknown.format === VCFormat.ldp_vc) {
+    const types = supportedCredentialsWellknown.credential_definition
+      .type as string[];
+    return types[1];
+  } else {
+    return i18n.t('VcDetails:nationalCard');
```

Comment:
fallback scenario for mso_mdoc = "Identity Card"
https://github.com/mosip/inji-wallet/pull/1705#discussion_r1861512414